libraries/AP_RangeFinder/AP_RangeFinder_MaxsonarI2CXL.cpp

```diff
@@ -22,25 +22,27 @@
  *
  *       Sensor should be connected to the I2C port
  */
-
 #include "AP_RangeFinder_MaxsonarI2CXL.h"
-#include <AP_HAL/AP_HAL.h>
+
 #include <utility>
 
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/utility/sparse-endian.h>
+
 extern const AP_HAL::HAL& hal;
 
-/* 
+/*
    The constructor also initializes the rangefinder. Note that this
    constructor is not called until detect() returns true, so we
    already know that we should setup the rangefinder
 */
-AP_RangeFinder_MaxsonarI2CXL::AP_RangeFinder_MaxsonarI2CXL(RangeFinder &_ranger, uint8_t instance, RangeFinder::RangeFinder_State &_state) :
-    AP_RangeFinder_Backend(_ranger, instance, _state),
-    _dev(hal.i2c_mgr->get_device(0, AP_RANGE_FINDER_MAXSONARI2CXL_DEFAULT_ADDR))
+AP_RangeFinder_MaxsonarI2CXL::AP_RangeFinder_MaxsonarI2CXL(RangeFinder &_ranger, uint8_t instance, RangeFinder::RangeFinder_State &_state)
+    : AP_RangeFinder_Backend(_ranger, instance, _state)
+    , _dev(hal.i2c_mgr->get_device(1, AP_RANGE_FINDER_MAXSONARI2CXL_DEFAULT_ADDR))
```

Comment:
Changing to 0 was a mistake. The original version used hal.i2c which for majority of boards is /dev/i2c-1. A more correct fix would be to add it to HAL_Boards.h but I didn't think it was necessary for this driver